binaries/llm-cli/src/main.rs

```diff
@@ -264,6 +266,33 @@ fn interactive(
             .unwrap_or(false)
     }
 
+    fn inference_callback(
```

Comment:
this is essentially a copy of the function here https://github.com/rustformers/llm/blob/main/crates/llm/examples/vicuna-chat.rs#L119